src/box/memtx_engine.c

```diff
@@ -230,6 +243,9 @@ memtx_engine_recover_snapshot_row(struct memtx_engine *memtx,
 	struct request request;
 	if (xrow_decode_dml(row, &request, dml_request_key_map(row->type)) != 0)
 		return -1;
+	if (is_not_system_space != NULL)
```

Comment:
I would always pass non-null pointer and assign is_not_system_space. Also I'd rename the latter to is_space_system. Finally, you can use space_is_system() function.